packages/react-vapor/src/components/select/hoc/tests/MultiSelectWithPredicate.spec.tsx

```diff
@@ -41,168 +26,81 @@ describe('Select', () => {
-            wrapper = mount(
-                <Provider store={store}>
-                    <MultiSelectWithPredicate {...basicProps} {...props} />
-                </Provider>,
-                {attachTo: document.getElementById('App')}
-            );
-            multiSelect = wrapper.find(SelectConnected).first();
-        };
-
-        beforeEach(() => {
-            store = TestUtils.buildStore();
-        });
-
-        afterEach(() => {
-            store.dispatch(clearState());
-            if (wrapper && wrapper.exists()) {
-                wrapper?.unmount();
-            }
-        });
-
-        describe('mount and unmount', () => {
-            it('should not throw on mount', () => {
-                expect(() => mountMultiSelect()).not.toThrow();
-            });
-
-            it('should not throw on unmount', () => {
-                mountMultiSelect();
+        it('hides items when they do not match the predicates', () => {
+            const items = [{value: 'first'}, {value: 'second'}, {value: 'third'}];
 
-                expect(() => wrapper.unmount()).not.toThrow();
-            });
-
-            it('should add the list box to the state when mounted', () => {
-                expect(store.getState().selects.length).toBe(0);
-
-                mountMultiSelect();
-
-                expect(store.getState().selects.length).toBe(1);
-            });
+            render(<MultiSelectWithPredicate {...basicProps} items={items} />, {});
+            // open the dropdown
+            userEvent.click(screen.getByRole('button', {name: 'Select an option'}));
 
-            it('should remove the list box from the state when the component unmount', () => {
-                mountMultiSelect();
+            let lists = screen.getAllByRole('list');
+            expect(within(lists[1]).getByText('first')).toBeInTheDocument();
 
-                expect(store.getState().selects.length).toBe(1);
-                wrapper.unmount();
+            // select the none flat select
+            userEvent.click(screen.getByText('None'));
 
-                expect(store.getState().selects.length).toBe(0);
-            });
-        });
-
-        it('should hide items when they do not match the predicates', () => {
-            const items = [{value: 'a'}, {value: 'b', selected: true}, {value: 'c', selected: true}];
-
-            mountMultiSelect({items});
-            store.dispatch(selectFlatSelect(id, defaultFlatSelectOptions[1].id));
-            wrapper.update();
-            multiSelect = wrapper.find(SelectConnected);
-
-            expect(multiSelect.props().items.length).toBe(items.length);
-            multiSelect
-                .find(SelectConnected)
-                .props()
-                .items.every((item: IItemBoxProps) => {
-                    expect(item.hidden).toBe(true);
-                });
+            lists = screen.getAllByRole('list');
+            expect(within(lists[1]).queryByText('first')).not.toBeInTheDocument();

```

Comment:
Why do we use `queryByText` here but `getByText` at line 37? Is it because `queryByText` is better for `.not.toBeInTheDocument()`?

Reply:
`getByText` will throw if the element is not in the document